builtin/logical/database/backend_test.go

```diff
@@ -185,53 +187,170 @@ func TestBackend_config_connection(t *testing.T) {
+	}
 	defer b.Cleanup(context.Background())
 
-	configData := map[string]interface{}{
-		"connection_url":    "sample_connection_url",
-		"plugin_name":       "postgresql-database-plugin",
-		"verify_connection": false,
-		"allowed_roles":     []string{"*"},
-	}
+	// Test creation
+	{
+		configData := map[string]interface{}{
+			"connection_url":    "sample_connection_url",
+			"someotherdata":     "testing",
+			"plugin_name":       "postgresql-database-plugin",
+			"verify_connection": false,
+			"allowed_roles":     []string{"*"},
+			"name":              "plugin-test",
+		}
 
-	configReq := &logical.Request{
-		Operation: logical.UpdateOperation,
-		Path:      "config/plugin-test",
-		Storage:   config.StorageView,
-		Data:      configData,
-	}
-	resp, err = b.HandleRequest(context.Background(), configReq)
-	if err != nil || (resp != nil && resp.IsError()) {
-		t.Fatalf("err:%s resp:%#v\n", err, resp)
-	}
+		configReq := &logical.Request{
+			Operation: logical.CreateOperation,
+			Path:      "config/plugin-test",
+			Storage:   config.StorageView,
+			Data:      configData,
+		}
 
-	expected := map[string]interface{}{
-		"plugin_name": "postgresql-database-plugin",
-		"connection_details": map[string]interface{}{
-			"connection_url": "sample_connection_url",
-		},
-		"allowed_roles":                      []string{"*"},
-		"root_credentials_rotate_statements": []string{},
+		exists, err := b.connectionExistenceCheck()(context.Background(), configReq, &framework.FieldData{
+			Raw:    configData,
+			Schema: pathConfigurePluginConnection(b).Fields,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+		if exists {
+			t.Fatal("expected not exists")
+		}
+
+		resp, err = b.HandleRequest(context.Background(), configReq)
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("err:%v resp:%#v\n", err, resp)
+		}
+
+		expected := map[string]interface{}{
+			"plugin_name": "postgresql-database-plugin",
+			"connection_details": map[string]interface{}{
+				"connection_url": "sample_connection_url",
+				"someotherdata":  "testing",
+			},
+			"allowed_roles":                      []string{"*"},
+			"root_credentials_rotate_statements": []string{},
+		}
+		configReq.Operation = logical.ReadOperation
+		resp, err = b.HandleRequest(context.Background(), configReq)
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("err:%s resp:%#v\n", err, resp)
+		}
+
+		delete(resp.Data["connection_details"].(map[string]interface{}), "name")
+		if !reflect.DeepEqual(expected, resp.Data) {
+			t.Fatalf("bad: expected:%#v\nactual:%#v\n", expected, resp.Data)
+		}
 	}
-	configReq.Operation = logical.ReadOperation
-	resp, err = b.HandleRequest(context.Background(), configReq)
-	if err != nil || (resp != nil && resp.IsError()) {
-		t.Fatalf("err:%s resp:%#v\n", err, resp)
+
+	// Test existence check and an update to a single connection detail parameter
+	{
+		configData := map[string]interface{}{
+			"connection_url":    "sample_convection_url",
```

Comment:
connection

Reply:
Nope, it's on purpose! Note that the previous check has `connection` and this has `convection`. The point is to check that it does indeed update, when other values that haven't been set stay the same.

Reply:
Perhaps this is too subtle, I just enjoyed changing it to another real word with a single char substitution.  :-)